### PR TITLE
feat(REST): RHICOMPL-1001 Count only supported results

### PR DESCRIPTION
**Pull request by akofink**
This aligns the REST API counts to the GraphQL counts for profile test
results. These attributes are used by the Insights Dashboard and must
match what GraphQL returns to make the dashboard consistent with the
Compliance reports page.

Test with https://github.com/RedHatInsights/insights-dashboard/pull/212

Signed-off-by: Andrew Kofink <akofink@redhat.com>

### CODE REVIEW

File: app/models/concerns/profile_scoring.rb [hunk 1]
```diff
@@ -4,6 +4,23 @@
 module ProfileScoring
   include ProfilePolicyScoring
 
+  def total_host_count
+    Host.where(id: assigned_hosts).or(Host.where(id: hosts)).count
```
[the remark this answers is not in the document]
Comment (akofink) [reply app/models/concerns/profile_scoring.rb:8]: Nice catch! I've added this case to the existing graphql profile query test, and it fails with this code as expected and passes with the new code.

File: test/graphql/queries/profile_query_test.rb [hunk 2]
```diff
@@ -274,7 +274,7 @@ class ProfileQueryTest < ActiveSupport::TestCase
     (host3 = hosts(:two).dup).update!(name: 'host3')
     policies(:one).update(compliance_threshold: 95,
                           account: accounts(:test),
-                          hosts: [hosts(:one), hosts(:two), host3])
+                          hosts: [hosts(:one), host3])
```
Comment (akofink) [inline test/graphql/queries/profile_query_test.rb:277]: Creating the scenario where we have test results for a host but it's not assigned to the policy.

File: test/graphql/queries/profile_query_test.rb [hunk 3]
```diff
@@ -285,8 +285,12 @@ class ProfileQueryTest < ActiveSupport::TestCase
     profile1_result = result['data']['allProfiles'].find do |h|
       h['id'] == profiles(:one).id
     end
+    profile2_result = result['data']['allProfiles'].find do |h|
+      h['id'] == profiles(:two).id
+    end
     assert_equal policies(:one).name, profile1_result['name']
-    assert_equal 3, profile1_result['totalHostCount']
+    assert_equal 2, profile1_result['totalHostCount']
+    assert_equal 2, profile2_result['totalHostCount']
```
Comment (akofink) [inline test/graphql/queries/profile_query_test.rb:293]: And, as expected, no matter which profile queried for `totalHostCount`, the result is the same.

**Review by vkrizan — APPROVED** [review]
Nice. Thank you.

**Pull request merged**